src/main/java/decodes/cwms/resevapcalc/ResWaterTemperatureCompute.java

```diff
@@ -0,0 +1,646 @@
+/* 
+ * Copyright (c) 2018
+ * United States Army Corps of Engineers - Hydrologic Engineering Center (USACE/HEC)
+ * All Rights Reserved.  USACE PROPRIETARY/CONFIDENTIAL.
+ * Source may not be released without written approval from HEC
+ */
+package decodes.cwms.resevapcalc;
+
+//import hec.heclib.util.HecTime;
+
+import java.io.BufferedWriter;
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ *  Class used to compute reservoir temperature profile
+ */
+public class ResWaterTemperatureCompute
+{
+    private static final Logger LOGGER = Logger.getLogger(ResWaterTemperatureCompute.class.getName());
+    BufferedWriter tout = null;
+
+    // reservoir layers, segments.
+    EvapReservoir reservoir;
+    int _resj_old = -1;
+
+    // working arrays
+    private double[] a;
+    private double[] b;
+    private double[] c;
+    private double[] r;
+    private double[] u;
+    private double[] rhow_org;
+    private double[] cp_org;
+    private double[] wt_tmp;
+    private double[] pe_mix_out;
+
+    public ResWaterTemperatureCompute(EvapReservoir reservoir )
+    {
+        this.reservoir = reservoir;
+        
+        // dimension working arrays
+        a = new double[EvapReservoir.NLAYERS];
+        b = new double[EvapReservoir.NLAYERS];
+        c = new double[EvapReservoir.NLAYERS];
+        r = new double[EvapReservoir.NLAYERS];
+        u = new double[EvapReservoir.NLAYERS];
+        rhow_org = new double[EvapReservoir.NLAYERS];
+        cp_org = new double[EvapReservoir.NLAYERS];
+        wt_tmp = new double[EvapReservoir.NLAYERS];
+        pe_mix_out = new double[EvapReservoir.NLAYERS];
+   }
+    
+    /**
+     * set output writer for diagnostic text output for reservoir 
+     * energy balance
+     *  
+     * @param tout
+     */
+    public void setOutfile( BufferedWriter tout )
```

Comment:
This should probably be a more generic input. Just `Writer` the caller can then decide if they want it buffered or not.